ppl/archive/import.go

```diff
@@ -116,9 +117,17 @@ func (w *Writer) Close() error {
 		}
 		delete(w.writers, ts)
 	}
+	for _, fn := range w.onclose {
+		fn()
+	}
 	return merr
 }
 
+// OnClose passes a function that will be called when the writer is closed.
```

Comment:
We don't use this pattern anywhere else, afaik. Instead of doing this, we would normally wrap a WriteCloser with another Closer. Is there a reason we can't use that instead of this approach?

Reply:
Sure